xbmc/utils/StringUtils.h

```diff
@@ -90,14 +91,129 @@ class StringUtils
-  static void ToLower(std::wstring &str);
+
+  /*!
+    * \brief Converts a string to Upper case according to locale.
+    *
+    * Note: This is a simplistic upper-case function which assumes that the string length
+    *       does not change during case change.
+    *
+    * \param str string to change case
+    * \param locale influences case changing rules
+    * \return Upper Case version of str
+    */
+  static std::string ToUpper(std::string_view str, std::locale locale = g_langInfo.GetSystemLocale());
+
+  /*!
+    * \brief Converts a wstring to Upper case according to locale.
+    *
+    * Note: This is a simplistic upper-case function which assumes that the string length
+    *       does not change during case change.
+    *
+    * \param str string to change case
+    * \param locale influences case changing rules
+    * \return Upper Case version of str
+    */
+  static std::wstring ToUpper(std::wstring_view str, std::locale locale = g_langInfo.GetSystemLocale());
+
+  /*!
+    * \brief Converts a string to Lower case according to locale.
+    *
+    * Note: This is a simplistic lower-case function which assumes that the string length
+    *       does not change during case change.
+    *
+    * \param str string to change case
+    * \param locale influences case changing rules
+    * \return Lower Case version of str
+    */
+  static std::string ToLower(std::string_view str, std::locale locale = g_langInfo.GetSystemLocale());
+
+  /*!
+    * \brief Converts a wstring to Lower case according to locale.
+    *
+    * Note: This is a simplistic lower-case function which assumes that the string length
+    *       does not change during case change.
+    *
+    * \param str string to change case on
+    * \param locale influences case changing rules
+    * \return Lower Case version of str
+    */
+  static std::wstring ToLower(std::wstring_view str, std::locale locale = g_langInfo.GetSystemLocale());
+
+   /*!
+    *  \brief Folds the case of a string using a simplistic algorithm.
+    *
+    * This is an embarrassingly simplistic version of FoldCase which simply performs a ToLower
+    * using the English locale (en_GB.UTF8). This should be sufficient to avoid problems like
+    * "the Turkish I" problem which occurs when ToLower is used with the Turkish locale. By specifying
+    * English, the character case rules are better understood and naturally follow the the primary use
+    * case for FoldCase, which is to 'normalize' the case of keywords used for table lookup, etc. As
+    * long as the keywords follow English rules (no accents, etc.) then this FoldCase should be sufficient.
+    *
+    * A more sophisticated FoldCase from ICU would be better, but the effort to integrate that into Kodi
+    * will not be able to be complete until another release.
+    *
+    * \param str string to fold
+    * \return Case folded version of str (all lower-case)
+    */
+
+  static std::string FoldCase(std::string_view str);
```

Comment:
i wondering if it is possible find a more appropriate name for these new methods
i find these names not immediately comprehensible